docs/scipp_widgets_intro.ipynb

```diff
@@ -0,0 +1,179 @@
+    "    'bob': sc.Variable(['x', 'z'], values=np.arange(0.0, 10.0, 0.1).reshape(10, 10),\n",
+    "                       variances=0.1*np.random.rand(10, 10))\n",
+    "    },\n",
+    "    coords={\n",
+    "        'x': sc.Variable(['x'], values=np.arange(11.0), unit=sc.units.m),\n",
+    "        'y': sc.Variable(['y'], values=np.arange(11.0), unit=sc.units.m),\n",
+    "        'z': sc.Variable(['z'], values=np.arange(11.0), unit=sc.units.m)})"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Wrapping a display function\n",
+    "\n",
+    "scipp-widgets allows functions to be wrapped in simple graphical interfaces. \n",
+    "As a first example, take the `scipp.plot.plot` function. This takes a scipp object and plots it. \n",
+    "In its simplest form, this only has one input; the object to be plotted. \n",
+    "\n",
+    "We will need to create an input specification object for this describing how we want it displayed, what validation we wish to perform and which argument of the underlying function it should correspond to.\n",
+    "\n",
+    "The minimum we need here is the name of the underlying function argument."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from scipp_widgets.input_spec import InputSpec\n",
+    "from scipp_widgets.widgets import DisplayWidget\n",
+    "data_input = InputSpec('scipp_obj', eval_input=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The `eval_input` allows variables in the notebooks global scope to be passed to this input by name.\n",
+    "\n",
+    "A widget which wraps the plot function can then be constructed like so:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "DisplayWidget(sc.plot.plot, 'Plot', [data_input])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Here the arguments are: function to wrap, name inside the button which triggers the execution, list of required input objects."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Wrapping a processing function\n",
+    "Another common case is to wrap a function which processes an input and produces an output. \n",
+    "This can be achieved using the `ProcessWidget` class. \n",
+    "As an example, we construct a wrapper around the `scipp.sum` function below."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from scipp_widgets.widgets import ProcessWidget\n",
+    "from scipp_widgets.validators import scipp_object_validator\n",
+    "data_input = InputSpec('x', tooltip='input data', eval_input=True)\n",
+    "dimension_input = InputSpec('dim', options = ('x', 'z'))\n",
```

Comment:
Ok for now, but this is not going to work in practice. We'd want to get the dims from the provided data. Will we need input widgets that are linked, or provide two input fields (for both name and dim)?

Reply:
Having input widgets which are linked adds quite a lot of additional complexity but agree it is something which we will want in the future. Not sure what the best way to solve this is currently so suggest we just note this down as a future issue to solve.